src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -501,9 +515,11 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 			return;
 		}
 		removedDocuments.forEach(removedDoc => {
-			let listener = this._modelToDisposeMap.get(removedDoc.toString());
-			if (listener) {
-				listener.dispose();
+			let listeners = this._modelToDisposeMap.get(removedDoc.toString());
+			if (listeners && listeners.length) {
```

Comment:
Having used this myself, I'm getting annoyed at the double check. I think we could use `Array.isArray` instead?